terraform/policies.tf

```diff
@@ -1,90 +1,98 @@
-{
-    "Version": "2012-10-17",
-    "Statement": [
-        {
-            "Effect": "Allow",
-            "Action": [
-                "iam:*LoginProfile",
-                "iam:*AccessKey*",
-                "iam:*SSHPublicKey*"
-            ],
-            "Resource": "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
-        },
-        {
-            "Effect": "Allow",
-            "Action": [
-                "iam:ListAccount*",
-                "iam:GetAccountSummary",
-                "iam:GetAccountPasswordPolicy",
-                "iam:ListUsers"
-            ],
-            "Resource": "*"
-        },
-        {
-            "Sid": "AllowUsersToCreateEnableResyncDeleteTheirOwnVirtualMFADevice",
-            "Effect": "Allow",
-            "Action": [
-                "iam:CreateVirtualMFADevice",
-                "iam:EnableMFADevice",
-                "iam:ResyncMFADevice",
-                "iam:DeleteVirtualMFADevice"
-            ],
-            "Resource": [
-                "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/$${aws:username}",
-                "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
+data "aws_iam_policy_document" "self_manage_iam_user" {
+    statement {
+        effect = "Allow"
+        actions = [
+            "iam:*LoginProfile",
+            "iam:*AccessKey*",
+            "iam:*SSHPublicKey*"
+        ]
+        resources = [
+            "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}",
+        ]
+    }
+
+    statement {
+        effect = "Allow"
+        actions = [
+            "iam:ListAccount*",
+            "iam:GetAccountSummary",
+            "iam:GetAccountPasswordPolicy",
+            "iam:ListUsers"
+        ]
+        resources = [
+            "*",
+        ]
+    }
+
+    statement {
+        sid = "AllowUsersToCreateEnableResyncDeleteTheirOwnVirtualMFADevice"
+        effect = "Allow"
+        actions = [
+            "iam:CreateVirtualMFADevice",
+            "iam:EnableMFADevice",
+            "iam:ResyncMFADevice",
+            "iam:DeleteVirtualMFADevice"
+        ]
+        resources = [
+            "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/$${aws:username}",
+            "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
+        ]
+    }
+
+    statement = {
+        sid = "AllowUsersToDeactivateTheirOwnVirtualMFADevice"
+        effect = "Allow"
+        actions = [
+            "iam:DeactivateMFADevice"
+        ]
+        resources = [
+            "arn:aws:iam::${data.aws_caller_identity.current.account_id}:mfa/$${aws:username}",
+            "arn:aws:iam::${data.aws_caller_identity.current.account_id}:user/$${aws:username}"
+        ]
+        condition {
+            test = "Bool"
+            variable = "aws:MultiFactorAuthPresent"
+            values = [
+                "true",
```

Comment:
I'm not confident that this is going to do the right thing: [using boolean values is discouraged](https://www.terraform.io/docs/configuration/variables.html#booleans) in favour of using strings instead, but this means that the constructed policy JSON has `"true"` instead of `true` for this value. (That's probably better than `"1"`, which is what comes out if we use `true` here.)